enhancements/installer/cluster-api-based-installations.md

```diff
@@ -0,0 +1,485 @@
+see-also:
+  - "/enhancements/this-other-neat-thing.md"
+replaces:
+  - "/enhancements/that-less-than-great-idea.md"
+superseded-by:
+  - "/enhancements/our-past-effort.md"
+---
+
+# Enabling Cluster-API-based Installations via openshift-install
+
+## Summary
+
+This enhancement discusses how `openshift-install` can use
+cluster-api (CAPI) providers to provision cluster infrastructure,
+without requiring access to an external management cluster or a local container runtime.
+By running a Kubernetes control plane and CAPI-provider controllers as
+subprocesses on the installer host, `openshift-install` can use CAPI and its
+providers in a similar manner to how Terraform and its providers are currently
+being used.
+
+## Motivation
+
+1. OpenShift Alignment with CAPI: CAPI offers numerous potential benefits;
+such as: day-2 infrastructure management, an API for users to edit cluster
+infrastructure, and upstream collaboration. Installer support for CAPI would
+be foundational for adopting these benefits.
+
+2. Terraform BSL License Change: due to the restrictive license change of
+Terraform, `openshift-install` needs a framework to replace the primary
+tool it used to provision cluster infrastructure. In addition to the benefits
+listed above, CAPI provides solutions for the biggest gaps left by Terraform:
+a common API for specifying infrastructure across cloud providers and robust
+infrastructure error handling. Furthermore, removing Terraform will greatly
+reduce the surface area for security vulnerabilities. Terraform and its providers
+constitutes over half a million lines of code in the Installer repo and is directly
+attributable for all (13) CVEs that have been fixed in the Installer repo.
+
+3. Streamline Installer development: a common pattern for Installer development
+has been to reimplement control-plane features in Terraform that have already been
+delivered upstream. By utilizing CAPI providers for Day-0 provisioning, our development
+practices will remove this duplication and become more efficient.
+
+### User Stories
+
+- As an existing user of the installer, I want to continue to use the installer binary (e.g. `create cluster`) in the same environments and with existing automation.
+- As an advanced user or cluster administrator, I want to be able to edit the CAPI infrastructure manifests so that I can customize control-plane infrastructure.
+
+
+
+### Goals
+
+- To provide a common experience across platforms for users and `openshift-install` developers
+- To be backwards compatible and fully satisfy the requirements of install-config type APIs.
+- To keep the user experience for day-zero operations unchanged or improved.
+- To not require any new runtime dependencies.
+- To maintain compatibility for hive, particularly regarding `destroy` functionality
+
+### Non-Goals / Future work
+
+- Non-goal: To maintain a consistent infrastructure footprint as clusters previously created with Terraform
```

Comment:
Is this non-goal saying that a cluster created with the installer cli after this proposal will produce a different infra than today? If so can we detail the differences and the impact on infra cost per cluster?

Reply:
The assumption this non-goal is making is that we're only promising API compatibility with the `InstallConfig`, rather than 1:1 compatibility between Terraform and Cluster API which in this case are primarily seen as implementation details.

In other words, there will be differences on how the infrastructure is created between the two systems, but the systems must be functionally equivalent. 

While 1:1 compatibility is a non-goal, we should add a small note that we should document as part of our release process the difference resulting infrastructure as best as possible.

Reply:
I think @enxebre 's point is that there are people and tools that rely on what the installer does today, so even though we can change the outputs and stay functionally equivalent we should be able to describe what will be different. Will names of resources change? Labels? Will a user see more or fewer of anything? Etc.

Reply:
Makes sense, that was my thinking as well; one of the graduation criteria should be proper documentation of the diff between old/new